cmd/kubeadm/app/cmd/phases/join/kubelet.go

```diff
@@ -114,6 +123,12 @@ func runKubeletStartJoinPhase(c workflow.RunData) (returnErr error) {
 
 	// Write the ca certificate to disk so kubelet can use it for authentication
 	cluster := tlsBootstrapCfg.Contexts[tlsBootstrapCfg.CurrentContext].Cluster
+
+	// If we're dry-running, write ca cert in tmp
+	if data.DryRun() {
+		cfg.CACertPath = filepath.Join(data.CertificateWriteDir(), kubeadmconstants.CACertName)
```

Comment:
i think we should avoid mutating configuration fields directy.
instead we could create a local variable

```
cfgPath = cfg.CACertPath
if data.DryRun() {
	cfgPath = filepath.Join(data.CertificateWriteDir(), kubeadmconstants.CACertName)
}
...
```

Reply:
good suggestion, fixed, thanks :)